dali/c_api_2/ref_counting.h

```diff
@@ -68,6 +88,11 @@ class RefCountedPtr {
     other.ptr_ = nullptr;
   }
 
+  RefCountedPtr &operator=(const RefCountedPtr &other) noexcept {
+    return this->operator= <T>(other);
+  }
+
```

Comment:
Out of curiosity, why do we need this when there is the templated variant below?